lib/screens/department_screen/widgets/faculty_details_section.dart

```diff
@@ -1,15 +1,353 @@
+import 'dart:convert';
+import 'dart:ui';
+
+import 'package:expandable/expandable.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:tsec_app/models/faculty_model/faculty_model.dart';
+import 'package:tsec_app/utils/themes.dart';
 
 class FacultyDetailsSection extends StatefulWidget {
-  const FacultyDetailsSection({Key? key}) : super(key: key);
-
+  const FacultyDetailsSection({Key? key, required this.department})
+      : super(key: key);
+  final String department;
   @override
   _FacultyDetailsSectionState createState() => _FacultyDetailsSectionState();
 }
 
 class _FacultyDetailsSectionState extends State<FacultyDetailsSection> {
+  late final Future<List<FacultyModel>> _faculties;
+
+  Future<List<FacultyModel>> _getFaculties() async {
+    String deptFileName;
+    switch (widget.department) {
+      case "Electronics &\nTelecommunication":
+        deptFileName = "extc";
+        break;
+      case "Biomedical":
+        deptFileName = "biomed";
+        break;
+      case "Biochemical":
+        deptFileName = "biochem";
+        break;
+      case "Chemical":
+        deptFileName = "chemical";
+        break;
+      case "Computer":
+        deptFileName = "cs";
+        break;
+      case "Information\nTechnology":
+        deptFileName = "it";
+        break;
+      case "First Year":
+        deptFileName = "fe";
+        break;
+      default:
+        deptFileName = "extc";
+    }
```

Comment:
Instead of a switch case use map here
They are computationally faster

Reply:
ok